app/models/mixins/authentication_mixin.rb

```diff
@@ -158,6 +158,8 @@ def update_authentication(data, options = {})
       cred.userid = value[:userid]
       cred.password = value[:password]
       cred.auth_key = value[:auth_key]
+      # TODO(lwander) this should have its own cred attribute
+      cred.password = value[:json] unless value[:json].blank?
```

Comment:
I can see this going either way.  I'm not gonna hold up the PR based on having or not having the new db field, but if you want to add it, it should be simple enough to drop the migration into this PR.

Even without the extra attribute, it probably makes sense to add an accessor method similar to [authentication_password](https://github.com/lwander/manageiq/blob/4162d4741715c0544190a0684d0744fc7f7c4ad0/app/models/mixins/authentication_mixin.rb#L45).  Something like:

```ruby
def authentication_service_account(type = nil)
  authentication_component(type, :password)
end
```

That way, [when you grab the service account json](https://github.com/ManageIQ/manageiq/pull/5208/files#diff-4ff38ff8b921da825c97c2006ece6d4bR50), you could call `authentication_service_account` and it will lend a little more context.  And, it doesn't really depend on the new db field.